packages/uicore/src/components/StringSubstitute/StringSubstitute.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Fragment } from 'react'
+
+type SubstituteVars = Record<string, any> // eslint-disable-line @typescript-eslint/no-explicit-any
+
+function translateExpression(str: string, key: string, vars: SubstituteVars) {
+  // Replace simple i18n expression {key|match1:value1,match2:value2}
+  // Sample: '{user} wants to merge {number} {number|1:commit,commits} into {target} from {source}'
+  // Find `{number|`
+  const startIndex = str.indexOf(`{${key}|`)
+  const MATCH_ELSE_KEY = '___'
+
+  if (startIndex !== -1) {
```

Comment:
*Minor* to avoid nested IFs, you could also do
```
if (startIndex === -1) {
    return str
}

// Find closing `}`
const endIndex = str.indexOf('}', startIndex)

if (endIndex === -1) {
    return str
}
```